src/views/undo.c

```diff
@@ -43,21 +44,30 @@ void dt_undo_cleanup(dt_undo_t *self)
   dt_pthread_mutex_destroy(&self->mutex);
 }
 
+static void _free_undo_data(void *p)
+{
+  dt_undo_item_t *item = (dt_undo_item_t *)p;
+  if (item->free_data) item->free_data(item->data);
+  free(item->data);
```

Comment:
I see, not really we need:
if (item->free_data) item->free_data(item->data);
free(item->data);
g_free(item);

Should be fixed now.